cibuildwheel/pyodide.py

```diff
@@ -65,11 +67,63 @@ def install_emscripten(tmp: Path, version: str) -> Path:
     return emcc_path
 
 
-def install_xbuildenv(env: dict[str, str], pyodide_version: str) -> str:
+def search_xbuildenv(env: dict[str, str]) -> list[str]:
+    """Searches for the compatible xbuildenvs for the current pyodide-build version"""
+    with FileLock(CIBW_CACHE_PATH / "xbuildenv.lock"):
+        xbuildenvs = call(
+            "pyodide",
+            "xbuildenv",
+            "search",
+            "--json",
+            "--all",
+            env=env,
+            cwd=CIBW_CACHE_PATH,
+            capture_stdout=True,
+        ).strip()
+        xbuildenvs_dict = json.loads(xbuildenvs)
+        compatible_xbuildenvs = [
+            env["version"] for env in xbuildenvs_dict["environments"] if env["compatible"]
+        ]
+        # Fetch just the "stable" versions
+        compatible_xbuildenvs_filtered = [
+            version for version in compatible_xbuildenvs if not any(_ in version for _ in "abc")
+        ]
+        # TODO: possibly remove that? Since this won't allow testing the unstable/dev versions
```

Comment:
I'm poised to remove this check, since this would allow downstream users to test versions such as [Pyodide 0.27.0a2](https://github.com/pyodide/pyodide/releases/tag/0.27.0a2) and see if things are working. Filtering the versions with this check restricts us within the same minor release at times (say, 0.26.2 or 0.26.3, etc.) and the Emscripten version bump/ABI break in Pyodide is never achieved until a new minor release, say, 0.27 or 0.28 (or later) is released. Testing with new Pyodide patch releases thus does not bring much merit.

Reply:
[`e551021`](https://github.com/pypa/cibuildwheel/pull/2002/commits/e551021a1bf71d8966baa6ace43adf162a6f7e23) removes this check, feedback is welcome.